crypto/kyber/p_kyber.c

```diff
@@ -109,6 +109,108 @@ static int pkey_kyber512_decapsulate(EVP_PKEY_CTX *ctx,
   return 1;
 }
 
+static int pkey_kyber768_keygen(EVP_PKEY_CTX *ctx, EVP_PKEY *pkey) {
```

Comment:
I thought a lot about ways to combine these methods. Do you think it's worth the effort to put this size specific data into the data pointer in EVP_PKEY_CTX like EC_PKEY_CTX which lets the EC code be generic across curves? I could be convinced the risk of combining this code and introducing a bug is higher than the chance we have a bug (or one of the "copies" gets of of date) is higher.

I think it could be worth it if we think there's a future where we have 4+ kems in AWS-LC and they could all use the same generic pkey_kem_keygen function instead of having 4 almost identical copies of these functions.

Reply:
I agree that redesigning the implementation as you suggest is worth only if we are going to support more KEMs. As long as we have only Kyber, I think it's ok to keep it simple and straightforward (on account of some code duplication). So let's discuss internally about this.